Sources/RoutingKit/Routing/TrieRouter.swift

```diff
@@ -70,11 +70,17 @@ public final class TrieRouter<Output>: CustomStringConvertible {
             // check the constants first
             for constant in currentNode.constants {
                 let match: Bool
-                if ci {
-                    match = constant.value.utf8.elementsEqual(caseInsensitive: path.utf8)
+                
+                // Short circuit early if items are different lengths
+                if constant.value.count != path.count {
+                    match = false
+                } else if ci {
+                    // constant.value will already be lowercased
+                    match = constant.value == path.lowercased()
```

Comment:
It's unfortunate that `path.lowercased()` is `O(n)` because we must do the count check to preserve some performance here. Previously, the lowercase value of the path was lazily calculated for each character as needed, meaning if only the first two characters of the string were matching, we only needed to calculate lowercase for the first three characters before detecting a mismatch.

Reply:
Yeah I agree. It'd be cool to get some sort of lazy lowercasing that would play well with `==`, but having looked at the equality implementation, idk how feasible it would be.